it/src/test/java/com/expediagroup/streamplatform/streamregistry/it/StreamRegistryIT.java

```diff
@@ -46,36 +43,34 @@
     ProducerBindingTestStage.class,
     ConsumerBindingTestStage.class
 })
+@Slf4j
 public class StreamRegistryIT {
-  private static final int POSTGRES_PORT = randomPort();
-
   public static Client client;
   public static String url;
 
   private static ConfigurableApplicationContext context;
 
   @ClassRule
-  public static FixedHostPortGenericContainer postgreSQLContainer =
-      new FixedHostPortGenericContainer<>("postgres:latest")
+  public static GenericContainer postgres =
+      new GenericContainer<>("postgres:12.0-alpine")
+          .withLogConsumer(o -> log.info("Postgres: {}", o.getUtf8String().trim()))
           .withEnv("POSTGRES_USER", "streamregistry")
           .withEnv("POSTGRES_PASSWORD", "streamregistry")
-          .withEnv("POSTGRES_DB", "streamregistry")
-          .withFixedExposedPort(POSTGRES_PORT, 5432);
+          .withEnv("POSTGRES_DB", "streamregistry");
 
   @BeforeClass
   public static void before() {
     String[] args = new String[]{
         "--server.port=0",
         "--schema.registry.url=http://schema-registry",
-        "--spring.datasource.url=jdbc:postgresql://localhost:" + POSTGRES_PORT + "/streamregistry",
+        "--spring.datasource.url=jdbc:postgresql://localhost:" + postgres.getMappedPort(5432) + "/streamregistry",
```

Comment:
Does this port get mapped into host's ports? Consider randomizing it!

Reply:
I was manually randomizing previously, but I discovered testcontainers already does it for you. `getMappedPort` returns you the random port that it assigned to the host port.

Reply:
it is also **highly** recommended to not use `localhost` here, see:
https://www.testcontainers.org/features/networking/

there is `container.getContainerIpAddress()` for that.

Reply:
If you use [the PostgreSQL module](https://www.testcontainers.org/modules/databases/postgres/), you can even replace it with `container.getJdbcUrl()`